src/data_access_layer/repositories/data_warehouse/data/edge_repository.ts

```diff
@@ -0,0 +1,480 @@
+            }
+
+            destination = request.value
+            e.destination_node_id = request.value.id!
+        } else {
+            return Promise.resolve(Result.Failure("no destination node id provided"))
+        }
+
+        if(e.metatypeRelationshipPair!.origin_metatype_id !== origin.metatype_id || e.metatypeRelationshipPair!.destination_metatype_id !== destination.metatype_id) {
+            return Promise.resolve(Result.Failure('origin and destination node types do not match relationship pair'))
+        }
+
+        // Note: At one point we also checked to see if the current relationship
+        // between the two proposed nodes existed. We thought that this would help
+        // cut down onA duplicates. However, it turns out that various services might
+        // create the same relationship between two nodes all the time - such as
+        // sensor data - and that attempting to tell apart what is an accidental
+        // edge creation vs. what is purposeful is doomed to fail. We have to put
+        // the burden on the users to ensure they're sending the right data, we can't
+        // have DL make assumptions about the data. However, we can still verify that
+        // we're not making new edges when doing so would violate a clause like one:one
+        // we just have to be careful how we build our query and to ignore something
+        // that already exits if it shares the same ID
+        let destinationQuery = new EdgeRepository() // new repository to avoid corrupting the filter
+            .where()
+            .destination_node_id("eq", destination.id!)
+            .and()
+            .relationshipPairID("eq", e.metatypeRelationshipPair!.id!)
+
+        let originQuery= new EdgeRepository() // new respository to avoid corrupting the filter
+            .where()
+            .origin_node_id("eq", origin.id!)
+            .and()
+            .relationshipPairID("eq", e.metatypeRelationshipPair!.id!)
+
+        if(e.composite_original_id) {
+            destinationQuery = destinationQuery.and().composite_original_id("eq", e.composite_original_id)
+            originQuery = originQuery.and().composite_original_id("eq", e.composite_original_id)
+        }
+
+        if(e.id) {
+            destinationQuery = destinationQuery.and().id("eq", e.id)
+            originQuery = originQuery.and().id("eq", e.id)
+        }
+
+        const destinationRelationships = await destinationQuery.count()
+        const originRelationships = await originQuery.count()
+
+        switch(e.metatypeRelationshipPair!.relationship_type) {
+            // we don't need to check a many:many as we don't have to verify more than whether or not the origin
+            // and destination types match
+            case "many:many": {
+                break;
+            }
+
+            case "one:one": {
+                if(!destinationRelationships.isError && destinationRelationships.value > 0 ) {
```

Comment:
Can you explain why this is > 0 and not > 1?

Reply:
If any current relationship exists between those nodes of the proposed types exist it would violate the `one:one` if we attempted to create a new relationship. Hence checking to see if any already exist. At least, that's the intention.